kubernetes/cache/ztunnel_dump.go

```diff
@@ -21,7 +30,37 @@ func (c *kialiCacheImpl) GetZtunnelDump(cluster, namespace, pod string) *kuberne
 	key := ztunnelDumpKey(cluster, namespace, pod)
 	config, found := c.ztunnelConfigStore.Get(key)
 	if !found {
-		return nil
+		if c.IsAmbientEnabled(cluster) {
+
+			ztunnelPods := c.GetZtunnelPods(cluster)
+			if len(ztunnelPods) > 0 {
+				client, err := c.GetKubeCache(cluster)
+				if err != nil {
+					log.Errorf("[GetZtunnelDump] Error getting kubecache for cluster %s: %v", cluster, err)
+				} else {
+					for _, zPod := range ztunnelPods {
+						if zPod.Name == pod {
+							resp, err := client.Client().ForwardGetRequest(zPod.Namespace, zPod.Name, 15000, "/config_dump")
+							if err != nil {
+								log.Errorf("[GetZtunnelDump] Error forwarding the /config_dump request: %v", err)
+								continue
+							}
+							configDump := &kubernetes.ZtunnelConfigDump{}
+							err = json.Unmarshal(resp, configDump)
+							if err != nil {
+								log.Errorf("[GetZtunnelDump] Error Unmarshalling the config_dump: %v", err)
+							} else {
+								key := fmt.Sprintf("%s%s%s", client.Client().ClusterInfo().Name, zPod.Namespace, zPod.Name)
+								zTunnelConfigStore := c.GetAllZtunnelDump()
+								zTunnelConfigStore[key] = configDump
+								c.SetZtunnelDump(zTunnelConfigStore)
+								return configDump
+							}
+						}
+					}
+				}
+			}
+		}
```

Comment:
minor nit that you don't need to change but you can probably rewrite most of this so it's not so nested which makes it easier to parse e.g.:
```
if found {
  return config
}

if !c.IsAmbientEnabled(cluster) {
  return nil
}

if len(ztunnelPod) <= 0 {
  return nil
}
...
```